lib/read_preference.js

```diff
@@ -16,19 +16,8 @@ const ReadPreference = function(mode, tags, options) {
     throw new TypeError(`Invalid read preference mode ${mode}`);
   }
 
-  // TODO(major): tags MUST be an array of tagsets
   if (tags && !Array.isArray(tags)) {
```

Comment:
Just a thought here: `tags` isn't optional so if you pass in tags as `false` or `null` or `undefined` (falsey values) this won't throw an error.

Reply:
Good point, I got rid of `tags &&` so it will always throw an error if it's not an array.

Reply:
Hey, I don't see this updated yet so can't approve yet.

Reply:
It turns out tags is actually optional, and that we want to support several construction styles:
```
constructor(selector, tags, options);
constructor(selector, options);
constructor(selector);
```
I've updated the jsdoc to show tags as being optional.